lib/events/events.go

```diff
@@ -314,19 +309,26 @@ func (s *bufferedSubscription) pollingLoop() {
 		s.buf[s.next] = ev
 		s.next = (s.next + 1) % len(s.buf)
 		s.cur = ev.SubscriptionID
-		s.cond.Broadcast()
+		if s.newEventNotification != nil {
+			close(s.newEventNotification)
+			s.newEventNotification = nil
+		}
 		s.mut.Unlock()
 	}
 }
 
-func (s *bufferedSubscription) Since(id int, into []Event) []Event {
+func (s *bufferedSubscription) Since(id int, into []Event, timeout time.Duration) []Event {
+	eventAvailable := s.waitForNewEventWithTimeout(id, timeout)
+	if !eventAvailable {
+		if into == nil {
+			return make([]Event, 0)
+		}
+		return into
+	}
+
```

Comment:
The JSON serializer doesn't think so - it serializes nil as emptystring, but an empty slice as `[]`. I wanted to return an empty JSON array in the case of no events, to be as friendly to consumers as possible. I'm happy to move this logic out into gui.go if you'd prefer.